src/v/raft/vote_stm.h

```diff
@@ -95,9 +95,19 @@ class vote_stm {
             // it is an error
             return state::error;
         }
+
+        bool has_longer_log() const {
+            return value && value->has_value() && !value->value().log_ok;
```

Comment:
nit: this is more like "more up-to-date log" (as the other candidate can have a log that is physically shorter, but has higher term entries at the end), right?